uaclient/clouds/identity.py

```diff
@@ -11,13 +13,32 @@
     # typing isn't available on trusty, so ignore its absence
     pass
 
+# Needed for Trusty-only
 CLOUDINIT_RESULT_FILE = "/var/lib/cloud/data/result.json"
+CLOUDINIT_INSTANCE_ID_FILE = "/var/lib/cloud/data/instance-id"
 
 
 # Mapping of datasource names to cloud-id responses. Trusty compat with Xenial+
 DATASOURCE_TO_CLOUD_ID = {"azurenet": "azure", "ec2": "aws"}
 
 
+def get_instance_id(
+    iid_file: str = CLOUDINIT_INSTANCE_ID_FILE
+) -> "Optional[str]":
+    """Query cloud instance-id from cmdline or CLOUDINIT_INSTANCE_ID_FILE"""
+    if util.which("cloud-id"):
```

Comment:
Using `cloud-id`s presence as a guard makes sense in `get_cloud_type` because it's then called there. Using its presence as a proxy for cloud-init version feels a little less obvious. Would either checking the Ubuntu release version, or explicitly checking for a lower-bound cloud-init version work instead?

Reply:
I agree here. Moving it to series-based trusty/non-trusty as we know that every other supported series has a recent enough cloud-init to have this query support. I had avoided calling cloud-init query specifically as trusty cloud-init query is a mangled/useless mess for that subcommand.